get_primary_services/device_code.js

```diff
@@ -0,0 +1,131 @@
+ *  limitations under the License.
+ */
+
+const data = new Uint32Array([3214]);
+let isConnected = false;
+let debug = '?';
+
+function updateScreen() {
+  if (typeof g === 'undefined') {
+    // Device has no graphics device (i.e. screen).
+    return;
+  }
+  g.clear();
+  g.setFontBitmap();
+  g.drawString('Current Notify Value:');
+  g.drawString('Connected: ' + isConnected, 0, g.getHeight() - 10);
+  let msg = 'Dbg: ' + debug;
+  g.drawString(msg, g.getWidth() - g.stringWidth(msg), g.getHeight() - 10);
+  g.setFontVector(40);
+  let val = data[0];
+  g.drawString(val, (g.getWidth() - g.stringWidth(val)) / 2, 12);
+
+  g.flip();
+}
+
+// This function is from:
+// https://stackoverflow.com/questions/105034/how-to-create-a-guid-uuid
+function generateUUID() {
+  return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function (c) {
+    var r = Math.random() * 16 | 0, v = c == 'x' ? r : (r & 0x3 | 0x8);
+    return v.toString(16);
+  });
+}
+
+function randomBoolean() {
+  return Math.random() < 0.5;
+}
+
+function createTestServices() {
+  const checkedCharacteristics = {
+    '3f2b9c10-7dce-11eb-9439-0242ac130002': {
+      value: data.buffer,
+      broadcast: false,
+      readable: true,
+      writable: false,
+      notify: true,
+      indicate: false,
+      description: 'Characteristic 1',
+    },
+    // Characteristic #2 has opposite values as #1.
+    'dcc030e4-8035-11eb-9439-0242ac130002': {
+      value: data.buffer,
+      broadcast: true,
+      readable: false,
+      writable: true,
+      notify: false,
+      indicate: true,
+      description: null,
+      onWrite : function(evt) {
+        console.log("Got ", evt.data);
+      }
+    }
+  };
+
+  // Add some more random characteristics. This is just for load testing to
+  // test of the browser can handle a reasonable number of characteristics.
+  //
+  // The nRF52832 (used by the Espruino) has limited resources. In testing
+  // will only list about six characteristics.
+  let uncheckedCharacteristics = {};
+  for (let i = 0; i < 10; i++) {
+    // Make the last two characters of the UUID the characteristic index. This
+    // is to make it easier to debug as the characters can be easily identified.
+    let digits = i.toString();
+    if (digits.length == 1)
+      digits = '0' + digits;
+    const uuid = generateUUID().replace(/..$/, digits);
+
+    uncheckedCharacteristics[uuid] = {
+      value: data.buffer,
+      broadcast: randomBoolean(),
+      readable: randomBoolean(),
+      writable: randomBoolean(),
+      notify: randomBoolean(),
+      indicate: randomBoolean(),
+      description: `Characteristic ${i}`
+    };
+  }
+
+  return {
+    '3f2b9742-7dce-11eb-9439-0242ac130002': checkedCharacteristics,
+    '6fc096cc-803b-11eb-9439-0242ac130002': uncheckedCharacteristics
+  };
+}
+
+function onInit() {
+  // Put into a known state.
+  digitalWrite(LED, isConnected);
+
+  NRF.setServices(createTestServices());
+
+  NRF.on('disconnect', (addr) => {
```

Comment:
I think you meant `reason` for the 'disconnect' event and `addr` for the 'connect' event.

Reply:
Thx. I'll fix the other tests in a follow-on PR.